### PR TITLE
Web 3444 add support for the new profile service endpoints survey

### PR DESCRIPTION
**Pull request by Vianney-VICTORIA**

### CODE REVIEW

File: src/Profile.js [hunk 1]
```diff
@@ -238,4 +238,24 @@ export default class Profile extends Service {
...
+   * @param   {Object} param
+   * @param   {Object} param.surveyData
+   * @returns {Promise}
+   */
+  addSurvey ({ surveyData } = {}) {
+    return this.fetch(
+      this.bearerTokenAuthHeader(),
+      this.userId === 0 ? '/api/v1/me/survey' : '/api/v1/users/' + this.userId + '/survey',
+      this.toBody({
+        surveyData: surveyData
```
Comment (ben-farrelly) [inline src/Profile.js]: In the backend this is actually `survey` - but I'm happy to change this when I pull it in

File: test/integration/ProfileTest.js [hunk 2]
```diff
@@ -250,4 +250,29 @@ describe('Profile Tests', function () {
...
+      function () {
+        swsClient.userId = 0
+        return swsClient.profile.addSurvey().then(
+          () => Promise.reject(new Error('Expected non-2xx HTTP response code')),
+          err => {
+            expect(err.httpStatus).not.to.equal(404)
+          }
+        )
+      }
+    )
```
Comment (ben-farrelly) [inline test/integration/ProfileTest.js]: These two tests are exactly the same.

File: test/integration/ProfileTest.js [hunk 3]
```diff
@@ -250,4 +250,29 @@ describe('Profile Tests', function () {
...
+    it(`confirms URI used in 'addSurvey()' method with user ID, by returning a non-404 HTTP response`,
+      function () {
+        swsClient.userId = 0
+        return swsClient.profile.addSurvey().then(
+          () => Promise.reject(new Error('Expected non-2xx HTTP response code')),
+          err => {
+            expect(err.httpStatus).not.to.equal(404)
+          }
+        )
+      }
```
Comment (GuillaumeBrousse) [inline test/integration/ProfileTest.js]: I think you could remove on both case `function(){` it's not useful there
And change `swsClient.userId = 0` (by any number different than the first test which is 0 to get two different test)

**Comment by matiaslg13** [comment]
Happy with the comments that @ben-farrelly and @GuillaumeBrousse  added. Besides that, looks good! Good stuff @Vianney-VICTORIA

File: test/integration/ProfileTest.js [hunk 4]
```diff
@@ -253,19 +253,7 @@ describe('Profile Tests', function () {
...
       function () {
-        swsClient.userId = 0
-        return swsClient.profile.addSurvey().then(
-          () => Promise.reject(new Error('Expected non-2xx HTTP response code')),
-          err => {
-            expect(err.httpStatus).not.to.equal(404)
-          }
-        )
-      }
-    )
```
Comment (GuillaumeBrousse) [inline test/integration/ProfileTest.js]: you should keep the first one as well with userId = 0

File: src/Profile.js (hunk 1 above)
Comment (Vianney-VICTORIA) [reply src/Profile.js]: yep Let's discuss about that !

File: test/integration/ProfileTest.js (hunk 2 above)
Comment (Vianney-VICTORIA) [reply test/integration/ProfileTest.js]: Good one :D , I 've removed one

File: test/integration/ProfileTest.js (hunk 3 above)
Comment (Vianney-VICTORIA) [reply test/integration/ProfileTest.js]: Fixed in the next commit

File: test/integration/ProfileTest.js (hunk 4 above)
Comment (Vianney-VICTORIA) [reply test/integration/ProfileTest.js]: Yep resolved in next commit

**Pull request merged**